contracts/SimpleYieldEulerLever.sol

```diff
@@ -0,0 +1,395 @@
+    //     |                                                    |                                                       Transfer eToken |            |         |
+    //     |                                                    |<----------------------------------------------------------------------|            |         |
+    //     |                                                    |                                                          |            |            |         |
+    //     |                                                    | pour x+y to borrow y fyToken                             |            |            |         |
+    //     |                                                    |----------------------------------------------------------------------------------->|         |
+    //     |                                                    |                                                          |            |            |         |
+    //     |                                                    | sell y to get underlying to payback the flashloan        |            |            |         |
+    //     |                                                    |--------------------------------------------------------------------------------------------->|
+    //     |                                                    |                                                          |            |            |         |
+    //     |                Transfer the vault back to the user |                                                          |            |            |         |
+    //     |<---------------------------------------------------|                                                          |            |            |         |
+    //     |                                                    |                                                          |            |            |         |
+    function invest(
+        bytes6 seriesId,
+        uint128 baseAmount,
+        uint128 borrowAmount,
+        uint128 minCollateral
+    ) external returns (bytes12 vaultId) {
+        // Transfer the tokens from user based on the ilk
+        IERC20(address(eToken)).safeTransferFrom(
+            msg.sender,
+            address(this),
+            baseAmount
+        );
+
+        // Build vault
+        (vaultId, ) = ladle.build(seriesId, ILKID, 0);
+
+        // Encode data of
+        // OperationType    1 byte      [0:1]
+        // seriesId         6 bytes     [1:7]
+        // vaultId          12 bytes    [7:19]
+        // baseAmount       16 bytes    [19:35]
+        // minCollateral    16 bytes    [35:51]
+        bytes memory data = bytes.concat(
+            bytes1(uint8(uint256(Operation.BORROW))),
+            seriesId,
+            vaultId,
+            bytes16(baseAmount),
+            bytes16(minCollateral)
+        );
+
+        bool success = assetJoin.flashLoan(
```

Comment:
If this is a flash loan of underlying, we can use [Euler as the flash loan provider](https://docs.euler.finance/developers/getting-started/integration-guide#eip-3156-flash-loans), which is convenient because they have more liquidity.

Reply:
Here are the gas costs when using our joins for flash loans
```
Running 1 test for test/YieldEulerLever.t.sol:ZeroStateTest
[PASS] testVault() (gas: 466235)
Test result: ok. 1 passed; 0 failed; finished in 9.58ms

Running 2 tests for test/YieldEulerLever.t.sol:UnwindTest
[PASS] testDoClose() (gas: 331333)
[PASS] testRepay() (gas: 352103)
Test result: ok. 2 passed; 0 failed; finished in 11.93ms
```
Here are the gas costs when using Euler provided flash loans
```
Running 1 test for test/YieldEulerLever.t.sol:ZeroStateTest
[PASS] testVault() (gas: 656390)
Test result: ok. 1 passed; 0 failed; finished in 141.52ms

Running 2 tests for test/YieldEulerLever.t.sol:UnwindTest
[PASS] testDoClose() (gas: 495634)
[PASS] testRepay() (gas: 352103)
Test result: ok. 2 passed; 0 failed; finished in 147.95ms
```

Using euler flash loans would ensure that liquidity is always available for any type of leverage but at a higher gas cost.